akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -55,74 +74,79 @@ abstract class Behavior[T] {
 
-  /**
-   * Narrow the type of this Behavior, which is always a safe operation. This
-   * method is necessary to implement the contravariant nature of Behavior
-   * (which cannot be expressed directly due to type inference problems).
-   */
-  def narrow[U <: T]: Behavior[U] = this.asInstanceOf[Behavior[U]]
 }
 
 object Behavior {
 
   /**
    * INTERNAL API.
    */
+  @InternalApi
   @SerialVersionUID(1L)
-  private[akka] object emptyBehavior extends Behavior[Any] {
-    override def management(ctx: ActorContext[Any], msg: Signal): Behavior[Any] = ScalaDSL.Unhandled
-    override def message(ctx: ActorContext[Any], msg: Any): Behavior[Any] = ScalaDSL.Unhandled
+  private[akka] object EmptyBehavior extends Behavior[Any] {
     override def toString = "Empty"
   }
 
   /**
    * INTERNAL API.
    */
+  @InternalApi
   @SerialVersionUID(1L)
-  private[akka] object ignoreBehavior extends Behavior[Any] {
-    override def management(ctx: ActorContext[Any], msg: Signal): Behavior[Any] = ScalaDSL.Same
-    override def message(ctx: ActorContext[Any], msg: Any): Behavior[Any] = ScalaDSL.Same
+  private[akka] object IgnoreBehavior extends Behavior[Any] {
     override def toString = "Ignore"
   }
 
   /**
    * INTERNAL API.
    */
+  @InternalApi
   @SerialVersionUID(1L)
-  private[akka] object unhandledBehavior extends Behavior[Nothing] {
-    override def management(ctx: ActorContext[Nothing], msg: Signal): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
-    override def message(ctx: ActorContext[Nothing], msg: Nothing): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
+  private[akka] object UnhandledBehavior extends Behavior[Nothing] {
+    // override def management(ctx: ActorContext[Nothing], msg: Signal): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
+    // override def message(ctx: ActorContext[Nothing], msg: Nothing): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
     override def toString = "Unhandled"
   }
 
   /**
    * INTERNAL API
    */
   @InternalApi private[akka] val unhandledSignal: (ActorContext[Nothing], Signal) ⇒ Behavior[Nothing] =
-    (_, _) ⇒ unhandledBehavior
+    (_, _) ⇒ UnhandledBehavior
 
   /**
    * INTERNAL API.
    */
+  @InternalApi
+  @DoNotInherit
   @SerialVersionUID(1L)
-  private[akka] object sameBehavior extends Behavior[Nothing] {
-    override def management(ctx: ActorContext[Nothing], msg: Signal): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
-    override def message(ctx: ActorContext[Nothing], msg: Nothing): Behavior[Nothing] = throw new UnsupportedOperationException("Not Implemented")
+  private[akka] abstract class DeferredBehavior[T] extends Behavior[T] {
+    /** "undefer" the deferred behavior */
+    @throws(classOf[Exception])
+    def apply(ctx: ActorContext[T]): Behavior[T]
+  }
```

Comment:
Delegate the actual deferred implementation to the respective DSLs to avoid problems with Java functions vs Scala functions